framework/include/utils/InputParameters.h

```diff
@@ -1830,3 +1851,79 @@ InputParameters::have_parameter(std::string_view name_in) const
 
   return Parameters::have_parameter<T>(name);
 }
+
+template <typename T>
+void
+InputParameters::transferParam(const InputParameters & source_params,
+                               std::string name_in,
+                               std::string new_description)
+{
+  const auto name = source_params.checkForRename(std::string(name_in));
+  if (!source_params.have_parameter<T>(name) && !source_params.hasCoupledValue(name))
+    mooseError(
+        "The '",
+        name_in,
+        "' parameter could not be transferred because it does not exist in the source parameters");
+
+  std::string description =
+      new_description.empty() ? source_params.getDescription(name) : new_description;
+
+  if (source_params.isParamRequired(name))
+  {
+    // Use the name_in that was specified in the transfer, not the potentially deprecated name
+    // Check for a variable parameter
+    if (source_params.hasCoupledValue(name))
+      addRequiredCoupledVar(name, description);
+    // Enums parameters have a default list of options
+    else if constexpr (std::is_same_v<MooseEnum, T> || std::is_same_v<MultiMooseEnum, T>)
+      addRequiredParam<T>(name_in, source_params.get<T>(name), description);
+    else if (source_params.isRangeChecked(name))
+      addRequiredRangeCheckedParam<T>(
+          name_in, source_params.rangeCheckedFunction(name), description);
+    else
+      addRequiredParam<T>(name_in, description);
+  }
+  else
+  {
+    // Check for a variable parameter
+    if (source_params.hasCoupledValue(name))
+    {
+      if (!source_params.hasDefaultCoupledValue(name))
+        addCoupledVar(name, description);
+      else if (source_params.numberDefaultCoupledValues(name) == 1)
+        addCoupledVar(name, source_params.defaultCoupledValue(name), description);
+      else
+      {
+        std::vector<Real> coupled_values;
+        for (const auto i : make_range(source_params.numberDefaultCoupledValues(name)))
+          coupled_values.push_back(source_params.defaultCoupledValue(name, i));
+        addCoupledVar(name, coupled_values, description);
```

Comment:
When adding the parameter, in most places you are using `name_in`, but here you are using `name`. If there is a special treatment for coupled variables vs. other parameters, it would be good to explain it

Reply:
Ok I ll take another look

Reply:
there is not. there is only bad copy pasting